dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,488 @@
+
+            score_future = client.submit(_score, model_meta_future, X_test,
+                                         y_test)
+            seq.add(score_future)
+
+    assert completed_jobs.keys() == model_meta_futures.keys()
+    return (params, model_meta_futures, history, list(completed_jobs.values()))
+
+
+class HyperbandCV(DaskBaseSearchCV):
+    """Find the best parameters for a particular model with cross-validation
+
+    This algorithm is state-of-the-art and only requires computational budget
+    as input. It will find close to the best possible parameters with the given
+    computational budget.
+
+    Parameters
+    ----------
+    model : object
+        An object that has support for ``partial_fit``, ``get_params``,
+        ``set_params`` and ``score``. This can be an instance of scikit-learn's
+        BaseEstimator
+    params : dict
+        The various parameters to search over.
+    max_iter : int, default=81
+        The maximum number of partial_fit calls to any one model.
+    eta : int, default=3
+        How aggressive to be in model tuning. It is not recommended to change
+        this value, and if changed we recommend ``eta=2`` or ``eta=4``. Higher
+        values implies high confidence in model selection.
+    random_state : int or np.random.RandomState
+        A random state for this class. Setting this helps enforce determinism.
+    scoring : str or callable
+        The scoring method by which to score different classifiers.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from dask_ml.model_selection import HyperbandCV
+    >>> from dask_ml.datasets import make_classification
+    >>> from sklearn.linear_model import SGDClassifier
+    >>>
+    >>> X, y = make_classification(chunks=20)
+    >>> est = SGDClassifier(tol=1e-3)
+    >>> params = {'alpha': np.logspace(-4, 0, num=1000),
+    >>>           'loss': ['hinge', 'log', 'modified_huber', 'squared_hinge'],
+    >>>           'average': [True, False]}
+    >>>
+    >>> search = HyperbandCV(est, params)
+    >>> search.fit(X, y, classes=np.unique(y))
+    >>> search.best_params_
+    ... {'loss': 'log', 'average': False, 'alpha': 0.0080502}
+
+    Attributes
+    ----------
+    cv_results_ : dict of lists
+        Information about the cross validation scores for each model.
+        All lists are ordered the same, and this value can be imported into
+        a pandas DataFrame. This dict has keys of
+
+            * ``rank_test_score``
+            * ``model_id``
+            * ``mean_fit_time``
+            * ``mean_score_time``
+            * ``std_fit_time``
+            * ``std_score_time``
+            * ``mean_test_score``
+            * ``std_test_score``
+            * ``partial_fit_calls``
+            * ``mean_train_score``
+            * ``std_train_score``
+            * ``params``
+            * ``param_value``
+            * ``mean_copy_time``
+
+    meta_ : dict
```

Comment:
Could you explain the difference between `meta_` and `history_`? Would it make sense for `meta_` and `history_` to be stored under the same attribute, perhaps under different keys in a dictionary?

Reply:
How 'bout this for a difference?

>    **meta_ : dict**: Information about every model that was trained. Can be used as input to ``fit_metadata``.
>    **history_ : list of dicts**: Information about every model after it is scored. Most models will be in here more than once because poor performing models are "killed" off early.

I try to explain that history will contain the same model more than once.

Reply:
Is there a better name for `meta_`?

Reply:
It can be passed into `HyperbandCV.fit_metadata`, `meta` or `metadata` will work best I think.